### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-08-31

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Data/Int/Interval.lean [hunk 1]
```diff
@@ -106,7 +106,11 @@ theorem card_Ico : (Ico a b).card = (b - a).toNat := (card_map _).trans <| card_
 @[simp]
 theorem card_Ioc : (Ioc a b).card = (b - a).toNat := (card_map _).trans <| card_range _
 
-@[simp]
+#adaptation_note
+/--
+Due to an error in nightly-2024-08-29, I've removed `@[simp]` here.
+Hopefully it can be restored, if needed, after nightly-2024-08-30.
```
Comment (TwoFX) [inline Mathlib/Data/Int/Interval.lean:112]: Given that "after nightly-2024-08-30" is now, what is the status here?

Comment (semorrison) [reply Mathlib/Data/Int/Interval.lean:112]: Cleaning up now!

File: Mathlib/Data/List/Chain.lean [hunk 2]
```diff
@@ -337,9 +337,10 @@ theorem Chain'.induction (p : α → Prop) (l : List α) (h : Chain' r l)
   unfold Chain' at h
   split at h
   · simp
-  · simp_all only [ne_eq, not_false_eq_true, head_cons, true_implies, mem_cons, forall_eq_or_imp,
-      true_and]
-    exact h.induction p _ carries initial
+  · simp_all only [ne_eq, head_cons, mem_cons, forall_eq_or_imp]
+    constructor
+    exact initial (by simp)
+    exact h.induction p _ carries (initial (by simp))
```
Comment (semorrison) [inline Mathlib/Data/List/Chain.lean:343]: I don't understand why this proof repair was needed.

Comment (TwoFX) [reply Mathlib/Data/List/Chain.lean:343]: Adding `reduceCtorEq` makes the original proof go through again.

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**